tests/test_tox_travis.py

```diff
@@ -45,20 +47,65 @@
 3.4 = other
 """
 
+tox_ini_travis_factors = b"""
+[tox]
+envlist = py{27,34,35}, docs
+
+[travis]
+language =
+    generic: py{27,35}, docs
+python =
+    2.7: py27, docs
+os =
+    osx: py{27,35}, docs
+"""
+
+tox_ini_travis_env = b"""
+[tox]
+envlist = py{27,35}-django{19,110}
+
+[travis:env]
+DJANGO =
+    1.9: django19
+    1.10: django110
+"""
+
+tox_ini_legacy_warning = b"""
+[tox]
+envlist = py{27,35}-django{19,110}
+
+[tox:travis]
+3.4 = py34, extra
+
+[travis:env]
+DJANGO =
+    1.9: django19
+    1.10: django110
+"""
+
 
 class TestToxTravis:
     def tox_envs(self):
         """Find the envs that tox sees."""
-        output = subprocess.Popen(
-            ['tox', '-l'], stdout=subprocess.PIPE).communicate()[0]
-        return output.decode('utf-8').strip().split()
+        returncode, stdout, stderr = self.tox_envs_raw()
+        assert returncode == 0, stderr
+        return [env for env in stdout.strip().split('\n')]
+
+    def tox_envs_raw(self):
+        """Return the raw output of finding what tox sees."""
+        proc = subprocess.Popen(
+            ['tox', '-l'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        stdout, stderr = proc.communicate()
+        return 0, stdout.decode('utf-8'), stderr.decode('utf-8')
```

Comment:
Should this not be `proc.returncode` instead of `0`? Otherwise, this assertion always passes.